examples/sha256-cairo-m/README.md

```diff
@@ -0,0 +1,87 @@
+# SHA-256 Cairo-M Implementation
+
+A complete SHA-256 cryptographic hash function implementation in Cairo-M with
+comprehensive Rust integration tests.
+
+## Project Structure
+
+- `cairom.toml` - Cairo-M project manifest file
+- `src/` - Cairo-M source files
+  - `sha256.cm` - SHA-256 implementation
+- `tests/` - Rust integration tests
+- `Cargo.toml` - Rust project configuration
+
+## Prerequisites
+
+### macOS Users
+
+You need to have LLVM and their LLD linker installed.
+
+```bash
+brew install llvm
+brew install lld
+```
+
+## Common Commands
+
+### Run all tests
+
+```bash
+cargo test
+```
+
+### Run a specific test
+
+```bash
+cargo test test_sha256_empty_string
+```
+
+### Show test output
+
+```bash
+cargo test -- --nocapture
+```
+
+Note: The required RUSTFLAGS are automatically configured in
+`.cargo/config.toml`
+
+## Implementation Details
+
+### SHA-256 Algorithm
+
+This implementation follows the NIST FIPS 180-4 specification for SHA-256:
+
+- Processes messages in 512-bit (64-byte) chunks
+- Uses 32-bit word operations compatible with Cairo-M's u32 type
+- Implements all required bitwise operations (rotate, shift, XOR, AND)
+- Supports messages up to 2 chunks (128 bytes) in the current configuration
```

Comment:
I think the real useful info here is our limitation on pre-padding and the size of input messages; can we make it clear somewhere?